SplunkRumWorkspace/SplunkRum/SplunkRum/AppStart.swift

```diff
@@ -42,39 +42,101 @@ private func processStartTime() throws -> Date {
 
+/*
+ This blog post https://eisel.me/startup explains how we can check the ProcessInfo's `ActivePrewarm` environment flag to know if the application launch
+ sequence was initiated due to prewarming. We track that scenario with `isPrewarm`.
+ 
+ This also noted here on this discussion regarding iOS startup timing:
+ https://github.com/MobileNativeFoundation/discussions/discussions/146
+ */
+var isPrewarm: Bool = false
+/*
+ According to https://developer.apple.com/documentation/uikit/app_and_environment/responding_to_the_launch_of_your_app/about_the_app_launch_sequence/ :
+ "In iOS 15 and later, the system may, depending on device conditions,
+ prewarm your app — launch nonrunning application processes to reduce the amount of time the user waits before the app is usable."
+ */
+var prewarmAvailable: Bool {
+    if #available(iOS 15.0, *) {
+        return true
+    }
+    return false
+}
+
+// Time interval (5 mins) chosen as a threshold for totally invalid calculations
+var possibleAppStartTimingErrorThreshold: TimeInterval = 60 * 5
+
+/*
+ Used to track whether the application came into the foreground during startup from a background state.
+ There are many reason's why an application's launch sequence might be initiated and the app not be brought to the foreground.
+ One example might be a user notification response that can be handled in the background. This action triggers the
+ didFinishLaunchingNotification. In a normal user initiated start, the application state when the
+ application is brought to the foreground is `inActive`. But in the notification response case, it will be `background`.
+ We can set this flag to do additional calculation adjustments.
+*/
+var wasBackgroundedBeforeWillEnterForeground: Bool = false
+
 func initializeAppStartupListeners() {
-    let events = [
-        UIApplication.didFinishLaunchingNotification,
-        UIApplication.willEnterForegroundNotification,
-        UIApplication.didBecomeActiveNotification
-    ]
-    var reportedEvents = Set<Notification.Name>()
-    events.forEach { event in
-        _ = NotificationCenter.default.addObserver(forName: event, object: nil, queue: nil) { (_) in
-            if !reportedEvents.contains(event) {
-                reportedEvents.insert(event)
-                if appStart == nil {
-                    return
-                }
-                appStart!.addEvent(name: event.rawValue)
-                if event == UIApplication.didBecomeActiveNotification {
-                    appStart!.end()
-                    OpenTelemetry.instance.contextProvider.removeContextForSpan(appStart!)
-                    appStart = nil
-                    // Because of heavy overlap and desired treatment of AppStart vs
-                    // ongoing app lifecycle stuff, initialize this now rather than
-                    // earlier to avoid double-reporting or more complex logic
-                    initializeAppLifecycleInstrumentation()
-                }
```

Comment:
I switched this up to use a closure for each so that we can remove the observers after we get the notification. Before these would stay alive for the whole session.